include/revng/Pipeline/Context.h

```diff
@@ -15,78 +15,73 @@
 #include "llvm/Support/MemoryBuffer.h"
 #include "llvm/Support/raw_ostream.h"
 
+#include "revng/Pipeline/Global.h"
 #include "revng/Pipeline/KindsRegistry.h"
-#include "revng/Pipeline/SavableObject.h"
 
 namespace pipeline {
 
-/// A simple pair that holds a savable object a name meant to describe a object
-/// which lifetime must be than the context.
-class NamedGlobalReference {
-private:
-  llvm::StringRef Name;
-  SavableObjectBase *Obj;
-
-public:
-  NamedGlobalReference(llvm::StringRef Name, SavableObjectBase &Obj) :
-    Name(Name), Obj(&Obj) {}
-
-public:
-  llvm::StringRef name() const { return Name; }
-  SavableObjectBase &value() const { return *Obj; }
-};
-
 /// A class that contains every object that has a lifetime longer than a
 /// pipeline.
 ///
 /// This includes the kinds and the NamedGlobals that will be available in that
 /// context.
 class Context {
 private:
-  llvm::StringMap<SavableObjectBase *> Globals;
+  GlobalsMap Globals;
+  llvm::StringMap<std::any> Anys;
```

Comment:
`Anys` is quite obscure and leaks an implementation detail.

Proposal: we have `Globals` that are mutable and must be `TupleTree`s, then we have `ImmutableGlobals` (was: `Anys` that can be of any type.

What's your take?

Let's doc this.

Reply:
the issue is that they are not immutable, they are non savable and non serializable. 

why does it leak implementation? it is privite and there is just a getter and a setter.

Reply:
In order to find a good name we need to identify the core of the difference between the two: one can be modified by the user and generates invalidation events, the other is just some global context for the benefit of pipes that the user is not interested in.

Proposals:
* `Globals` -> `MutableGlobals` and `Anys` -> `ImmutableGlobals`
* `Anys` -> `ImmutableGlobals`
* `Anys` -> `PrivateGlobals`
* `Anys` -> `Contexts`
* `Anys` -> `NamedContexts`
* `Globals` -> `Configurations` and `Anys` -> `Contexts`

Reply:
Uh, Contexts is not a bad pick.

* Anys -> Contexts (addExternalContext, getExternalContext)

with this globals, can stay named globals